File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/c_solutions/1615-maximal-network-rank.c
int maximalNetworkRank(int n, int** roads, int roadsSize, int* roadsColSize) {
    int degree[n];
    int connected[n][n];
    memset(degree, 0, sizeof(degree));
    memset(connected, 0, sizeof(connected));

    for (int i = 0; i < roadsSize; i++) {
        int a = roads[i][0];
        int b = roads[i][1];
        degree[a]++;
        degree[b]++;
        connected[a][b] = 1;
        connected[b][a] = 1;
    }

    int maxRank = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            int rank = degree[i] + degree[j];
            if (connected[i][j]) {
                rank--;
            }
            if (rank > maxRank) {
                maxRank = rank;
            }
        }
    }

    return maxRank;
}

File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/erlang_solutions/1615-maximal-network-rank.erl
-spec maximal_network_rank(N :: integer(), Roads :: [[integer()]]) -> integer().
maximal_network_rank(N, Roads) ->
    Degrees = lists:foldl(fun([U, V], Acc) ->
                                  maps:update_with(U, fun(D) -> D + 1 end, 1, 
                                  maps:update_with(V, fun(D) -> D + 1 end, 1, Acc)))
                          end, maps:new(), Roads),
    Max = 0,
    Cities = lists:seq(0, N - 1),
    lists:foldl(fun(I, CurrentMax) ->
        lists:foldl(fun(J, InnerMax) ->
            if 
                I == J -> InnerMax;
                true ->
                    DegreeI = maps:get(I, Degrees, 0),
                    DegreeJ = maps:get(J, Degrees, 0),
                    Connected = case lists:member([I, J], Roads) orelse lists:member([J, I], Roads) of
                                    true -> 1;
                                    false -> 0
                                end,
                    Rank = DegreeI + DegreeJ - Connected,
                    max(InnerMax, Rank)
            end
        end, CurrentMax, Cities)
    end, Max, Cities).